src/OrchardCore/OrchardCore.DisplayManagement/Theming/ThemingViewsFeatureProvider.cs

```diff
@@ -13,7 +14,12 @@ namespace OrchardCore.DisplayManagement.Theming
     /// </summary>
     public class ThemingViewsFeatureProvider : IApplicationFeatureProvider<ViewsFeature>
     {
-        public static readonly string ThemeLayoutFileName = "DefaultOrchardCoreThemingLayout" + RazorViewEngine.ViewExtension;
+        public static readonly string ThemeLayoutFileName = $"DefaultOrchardCoreThemingLayout{RazorViewEngine.ViewExtension}";
+        public static readonly string ThemeLayoutFilePath = $"/{ThemeLayoutFileName}";
+        public static readonly string MvcViewDocumentKind = $@"""{MvcViewDocumentClassifierPass.MvcViewDocumentKind}""";
```

Comment:
It doesn't look like you understood my last comment. The generated code should point to the static field. Here you are sill using the value of the field `"mvc.1.view"` in the generated code and each view instance would have a distinct string instance. Not any different than before the PR. Here is what I have in mind, please check how the generated code is different (or at least should if I didn't make a mistake).

```suggestion
        public static readonly string MvcViewDocumentKind = "\"MvcViewDocumentClassifierPass.MvcViewDocumentKind\"";
```

Reply:
Okay, I will do the change.

For info it works whatever we pass, even if we pass null.

I will investigate more in depth asap.

Reply:
It's possible that this string is actually never used ;)

Reply:
We spent more time on the issue than we saved bytes probably, but I think it's an important behavior to understand.

Reply:
Yes I want to understand too, I will look at it when I will have time